src/zkbob/ZkBobCompoundingMixin.sol

```diff
@@ -0,0 +1,240 @@
+     */
+    function rebalance(uint256 minRebalanceAmount, uint256 maxRebalanceAmount) external {
+        if (maxRebalanceAmount < minRebalanceAmount) {
+            (minRebalanceAmount, maxRebalanceAmount) = (maxRebalanceAmount, minRebalanceAmount);
+        }
+
+        YieldParams storage params = yieldParams;
+        IERC4626 yieldVault = IERC4626(params.yield);
+
+        uint256 currentDust = params.dust;
+        if (currentDust > maxRebalanceAmount) {
+            return;
+        }
+        if (minRebalanceAmount < currentDust) {
+            minRebalanceAmount = currentDust;
+        }
+
+        if (address(yieldVault) == address(0)) {
+            return;
+        }
+
+        address operator = params.yieldOperator;
+        require(
+            operator == address(0) || operator == msg.sender,
+            "ZkBobCompoundingPool: Rebalance is an operator-called method"
+        );
+
+        uint256 underlyingBalance = IERC20(token).balanceOf(address(this));
+        uint256 buffer = params.buffer;
+
+        uint256 balancesDiff = (underlyingBalance >= buffer) ? underlyingBalance - buffer : buffer - underlyingBalance;
+        balancesDiff = (balancesDiff >= minRebalanceAmount)
+            ? ((balancesDiff <= maxRebalanceAmount) ? balancesDiff : maxRebalanceAmount)
+            : 0;
+        if (balancesDiff <= params.dust) {
+            return;
+        }
+        uint256 vaultAssets = investedAssetsAmount;
+        uint256 maxInvestedAmount = params.maxInvestedAmount;
+        if (underlyingBalance > buffer) {
+            if (vaultAssets + balancesDiff > maxInvestedAmount) {
+                balancesDiff = maxInvestedAmount - vaultAssets;
+                if (balancesDiff < minRebalanceAmount) {
+                    return;
+                }
+                investedAssetsAmount = maxInvestedAmount;
+            } else {
+                investedAssetsAmount += balancesDiff;
+            }
+            yieldVault.deposit(balancesDiff, address(this));
+            emit Rebalance(address(yieldVault), 0, balancesDiff);
+        } else {
+            if (balancesDiff > vaultAssets) {
+                balancesDiff = vaultAssets;
+                if (balancesDiff < minRebalanceAmount) {
+                    return;
+                }
+                investedAssetsAmount = 0;
+            } else {
+                investedAssetsAmount -= balancesDiff;
+            }
+            yieldVault.withdraw(balancesDiff, address(this), address(this));
+            emit Rebalance(address(yieldVault), balancesDiff, 0);
+        }
+    }
+
+    /**
+     * @dev Collects accumulated fees and generated yield for the specific collateral.
+     * Callable only by the contract owner / proxy admin / yield admin.
+     * @param minClaimAmount minimum amount of token to claim.
+     * @return Claimed amount.
+     */
+    function claim(uint256 minClaimAmount) external returns (uint256) {
+        YieldParams storage params = yieldParams;
```

Comment:
I believe it's better to read the whole struct into memory at once, if all slots are needed anyway.
In other cases, it's better to read fields allocated to the same slot as tuples:
```solidity
YieldParams storage params = yieldParams;
(a, b) = (params.a, params.b);
```

I think this is relevant to multiple places in the code.